cmd/thanos/compact.go

```diff
@@ -302,9 +305,20 @@ func runCompact(
 			cancel()
 		}
 	}()
+
+	var mergeFunc storage.VerticalChunkSeriesMergeFunc
+	switch conf.dedupFunc {
+	case compact.DedupAlgorithmPenalty:
+		mergeFunc = dedup.NewChunkSeriesMerger()
+
+		if len(conf.dedupReplicaLabels) == 0 {
+			return errors.New("penalty based deduplication needs at least one replica label specified")
+		}
```

Comment:
Can we error out on default, so if someone puts something not supported?

Reply:
Good idea!
But How can we detect unsupported labels? As it is experimental so users have to take the risk and backup data first?

Reply:
@yeya24 I think Bartek talks about dedupFunc value.

Reply:
I see. 😓